### PR TITLE
feat(connector) : add PayPal wallet support for Paypal

### PR DESCRIPTION
**Pull request by prasunna09**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Add PayPal wallet support for Paypal.

### Additional Changes

- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

tested using postman.
Screenshot 2023-04-17 at 10 53 40 AM

Screenshot 2023-04-17 at 10 54 02 AM

Screenshot 2023-04-17 at 10 54 39 AM

Screenshot 2023-04-17 at 10 55 30 AM
Screen Shot 2023-04-20 at 2 21 33 PM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/core/payments/transformers.rs [hunk 1]
```diff
@@ -676,11 +681,22 @@ impl<F: Clone> TryFrom<PaymentAdditionalData<'_, F>> for types::CompleteAuthoriz
...
             .map(|b| b.parse_value("BrowserInformation"))
             .transpose()
             .change_context(errors::ApiErrorResponse::InvalidDataValue {
                 field_name: "browser_info",
             })?;
+        let connector = api::ConnectorData::get_connector_by_name(
+            &additional_data.state.conf.connectors,
+            &additional_data.connector_name,
+            api::GetToken::Connector,
+        );
```
Comment (ArjunKarthik) [inline crates/router/src/core/payments/transformers.rs]:
```suggestion
        let connector = api::ConnectorData::get_connector_by_name(
            &additional_data.state.conf.connectors,
            &additional_data.connector_name,
            api::GetToken::Connector,
        )?;
```

By this way we can ignore the below match statement

File: crates/router/src/core/payments/transformers.rs [hunk 2]
```diff
@@ -676,11 +681,22 @@ impl<F: Clone> TryFrom<PaymentAdditionalData<'_, F>> for types::CompleteAuthoriz
...
+        let connectors = match connector {
```
Comment (ArjunKarthik) [inline crates/router/src/core/payments/transformers.rs]: This can be removed

File: crates/router/src/connector/paypal/transformers.rs [hunk 3]
```diff
@@ -111,6 +130,35 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for PaypalPaymentsRequest {
...
+                        }));
+
+                    Ok(Self {
+                        intent,
+                        purchase_units,
+                        payment_source,
+                    })
+                }
+                _ => Err(errors::ConnectorError::NotImplemented(
+                    "Unknown Wallet in Payment Method".to_string(),
```
Comment (ArjunKarthik) [inline crates/router/src/connector/paypal/transformers.rs:159]: Let it be `Payment Method`.

File: crates/router/src/connector/paypal/transformers.rs [hunk 4]
```diff
@@ -328,6 +391,47 @@ impl<F, T>
...
+    fn try_from(
+        item: types::ResponseRouterData<F, PaypalRedirectResponse, T, types::PaymentsResponseData>,
+    ) -> Result<Self, Self::Error> {
+        let status = storage_enums::AttemptStatus::foreign_from((
+            item.response.status,
+            item.response.intent.clone(),
+        ));
+        let mut link: Option<Url> = None;
+        let link_vec = item.response.links;
+        for item2 in link_vec.iter() {
```
Comment (ArjunKarthik) [inline crates/router/src/connector/paypal/transformers.rs:408]: Let's move this logic to a new method, `get_redirect_url`

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

**Comment by ArjunKarthik** [comment]
@prasunna09 You conflicts resolution removed some of your changes. Please verify your changes once again and update the PR with necessary changes

**Comment by ArjunKarthik** [comment]
@prasunna09 please resolve conflicts

**Review by ArjunKarthik — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Pull request merged**